From 21d6217d2ad7b1355b532623ad37fddd894b511b Mon Sep 17 00:00:00 2001
From: CI Assistant <chdauto@intel.com>
Date: Mon, 23 Mar 2020 13:53:20 +0200
Subject: [PATCH] Merge pull request #1938 in SW_UGW/linux from
 bugfix/UGW_SW-47745-multicast-functionality-broken-in-bridged-mode-8.4.1 to
 8.4.1

* commit 'fa684b3e83efbefc88a60b33ef81ad66e684ca8f':
  UGW_SW-47745: Allow generic igmp query packets in bridged mode
---
 net/bridge/ltq_br_mcast_snooping.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/net/bridge/ltq_br_mcast_snooping.c b/net/bridge/ltq_br_mcast_snooping.c
index 594e96ed5560..f429841dc681 100644
--- a/net/bridge/ltq_br_mcast_snooping.c
+++ b/net/bridge/ltq_br_mcast_snooping.c
@@ -620,6 +620,8 @@ static int br_selective_flood4(struct net_bridge_port *p, struct sk_buff *skb)
 		case IGMP_HOST_MEMBERSHIP_QUERY:
 			if (p->igmp_router_port)
 				return 0;
+			if ((ntohl(iph->daddr) & 0xffffffff) == 0xe0000001)
+				return 1;	/*Allow general query packets */
 			igmph3 = (struct igmpv3_query *)igmph;
 			igmplen = ntohs((int)iph->tot_len) - 4 * (int)iph->ihl;
 			if (igmplen == 8) {
